### PR TITLE
regression

### LINKED ISSUE #20: Visualizations for Regression-based NNs
Today all the visualizations are for classification. Let's extend the APIs to visualize regression based models

### PR DESCRIPTION
**Pull request by mcxxmc**
**Description of the change:**

This PR enables regression for 2d input; it also updates the description for render().

**Motivation for the change:**

To enable visualization for regression.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient test coverage 
- [ ] Docs updated
- [ ] Commit messages sensible and descriptive

Closes #20

### CODE REVIEW

**Review by gallettilance — COMMENTED** [review]
Looks great! Can you add a couple examples to the README? Feel free to use something like this:
https://medium.com/@gallettilance/introduction-to-neural-networks-intuition-from-linear-regression-c494fd4661cd

**Comment by mcxxmc** [comment]
of course, but I think I will make it a seperate PR because I am thinking about something different in this PR. More exactly, I am curious about what the visualization for the input data should look like.
Currently my code works for a 2-d input, and it visualizes the 2 dimensions of the input data but not the output (if output is included the visualization should be a 3d image). 
And in your example from the link, the visulization is about the 1-d input and its output (so the final image is 2d). I am wondering what kind of visualization is needed, since the input by user can in fact be of any dimensions. Note that we do not need to worry about the vis for node activations, since it will always be 2d and the difference is only about number of nodes in each layer.
I think a good idea is to make visualization for different kind of user input, depending on the dimensions. However, since we are living in a 3d world a visualization of input more than 2d (if including the output) or 3d (without output) is not possible. One way is not to visualize input for high dimensions, the other is to introduce some kind of mechnism to cut the input to lower dimensions (e.g., project on a 2d plane with highest variance; but that may involve complex computation)
What do you think?
Also, I can try to squash the commits after all changes are done, so (if I am lucky) there will be only a commit to be merged. If you prefer a single commit instead of a series of commits, please let me know before the merge. Thanks!

**Comment by mcxxmc** [comment]
hi professor @gallettilance could you please take a look at my comments above? thanks

File: kviz/dense.py [hunk 1]
```diff
@@ -490,6 +563,101 @@ def render(self, filename='graph'):
...
+                        index = unique_index(l + 1, h)
+
+                        set_node_attributes(self._graph, {
+                            index: {
+                                'label': label,
+                                'fixedsize': True
+                            }})
+
+                network_images.append(self._snap(filename))
+                self._update_input_images_for_regression(input_images, [i], X, predictions[-1], x_color, x_marker)
```
Comment (gallettilance) [inline kviz/dense.py:647]: what happens when the input is more than 2d? is there a nice error somewhere or does it just not show the input images?

**Review by gallettilance — COMMENTED** [review]
Looks really good!!

File: kviz/dense.py (hunk 1 above)
Comment (mcxxmc) [reply kviz/dense.py:699]: Updated. 
I think this should not be an error, so it does not show the input image if > 2d. The final gif will still have node activations.
For future versions maybe we can use projection (for example, tsne?) to visulize high dimension inputs.

**Review by gallettilance — APPROVED** [review]
lgtm!

**Pull request merged**